### PR TITLE
Replace truthy check with isset when setting default checkout field value

### PR DESCRIPTION
**Pull request by jkudish**
<details>
<summary>### Submission Review Guidelines:</summary>

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

</details>

### Changes proposed in this Pull Request:

This allows a falsey checkout field value to be set. Without this change, you aren't able to use the `woocommerce_get_default_value_for_{key}` filter to set a falsey value on a checkbox (i.e. to uncheck it dynamically) 

Fixes the issue i reported in https://github.com/woocommerce/woocommerce/issues/48030

### How to test the changes in this Pull Request:

Here's the simplest replication scenario:

Add the following code snippet to your site

```php
add_action( 'woocommerce_blocks_loaded', function(){
	woocommerce_register_additional_checkout_field( [  
	    'id'       => 'my_namespace/my_field',  
	    'label'    => 'Custom Checkbox',  
	    'location' => 'contact',  
	    'type'     => 'checkbox',  
	] );  
  
	// get the value from cart data  
	add_filter("woocommerce_get_default_value_for_my_namespace/my_field",  
    function ( $value ) {  
		return rand(0,1); // simulate a dynamic truthy/falsey value
    } );
} );
```

Then load your checkout page and see the "custom checkbox". Refresh several times, ensure you see the checkbox both checked and unchecked.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Fix a bug with the `woocommerce_get_default_value_for_{key}` filter that was preventing setting a falsey value on a checkbox (i.e. to uncheck it dynamically) 

</details>

</details>

### CODE REVIEW

**Comment by senadir** [comment]
Approved tests, let's see if this breaks anything, I assume it doesn't.

File: plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php [hunk 1]
```diff
@@ -1187,7 +1187,7 @@ public function get_all_fields_from_object( WC_Data $wc_object, string $group =
 				 */
 				$value = apply_filters( "woocommerce_get_default_value_for_{$missing_field}", null, $group, $wc_object );
 
-			if ( $value ) {
+			if ( isset($value) ) {
```
Comment (opr) [inline plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php]:
```suggestion
			if ( isset( $value ) ) {
```

Comment (senadir) [inline plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFields.php]:
```suggestion
			if ( isset( $value ) ) {
```

**Review by opr — CHANGES_REQUESTED** [review]
Hi @jkudish thanks a lot for the PR! The change makes sense, but just a few small things to improve:

- We should ensure the [WP Coding Standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#space-usage) are being followed
- Please could you write a line for the changelog? You can do `pnpm --filter=@woocommerce/plugin-woocommerce changelog add` on the command line to get this file automatically generated and then commit that.
- While the reproduction steps in https://github.com/woocommerce/woocommerce/pull/48031 are good, it would be really helpful to have the steps copied here as well (just makes it that little bit easier for testers)

Again, thanks for the submission, and let us know if you need a hand with any of the above 🙇🏼

**Comment by senadir** [comment]
This PR needs a change log, the initial template should have included a checkbox and message place to auto generate that.

**Comment by jkudish** [comment]
Hi @opr and @senadir,

Thanks for the quick responses, much appreciated!

> This PR needs a change log, the initial template should have included a checkbox and message place to auto generate that.

My apologies - i think I had accidentally removed that when creating the PR. I've re-added that now and the github actions picked up on it but sadly still failed:

image

> Please could you write a line for the changelog? You can do pnpm --filter=@woocommerce/plugin-woocommerce changelog add on the command line to get this file automatically generated and then commit that.

I was able to do it with this method. Changelog file added 👍🏼  


> We should ensure the [WP Coding Standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#space-usage) are being followed
> While the reproduction steps in https://github.com/woocommerce/woocommerce/pull/48031 are good, it would be really helpful to have the steps copied here as well (just makes it that little bit easier for testers)

Both also taken care of 👍🏼

**Comment by jkudish** [comment]
Hi @opr and @senadir, anything else required here to get this merged? :)

**Review by opr — APPROVED** [review]
Thanks @jkudish this is looking good now, I made a small adjustment to the testing steps. Really appreciate your PR and patience on this! 🙇🏼

**Comment by jkudish** [comment]
Awesome, thank you @opr ! :)

**Comment by jkudish** [comment]
@opr Thank you again for merging this!

Do you know which version of WooCommerce this will end up in? 8.9.3? 9.0.0-rc2? 9.1? I'll need to make the new code that we're using this in be aware of the version this is fixed in :)

No worries if that's unknown for now, I can keep an eye out, but thought I'd ask!

Cheers

**Comment by opr** [comment]
@jkudish no problem! It will ship with WooCommerce 9.1 which is scheduled for release on 9th July

If you do another PR in future and have the same question, the Milestone value on the PR's sidebar will let you know what release it will be part of.

image

**Comment by jkudish** [comment]
Ah perfect, I missed that will know for next time, thank you again @opr  :)

**Pull request merged**